collect_app/src/main/java/org/odk/collect/android/logic/FormController.java

```diff
@@ -833,6 +833,10 @@ public void setLanguage(String language) {
         return questions;
     }
 
+    private boolean isGroupEmpty() {
+        return getIndicesForGroup((GroupDef) formEntryController.getModel().getForm().getChild(getFormIndex())).isEmpty();
```

Comment:
This line is a bit much, mind factoring out<br/>`GroupDef group = (GroupDef) formEntryController.getModel().getForm().getChild(getFormIndex());` ?

Reply:
Improved.